### PR TITLE
SA-2022 - Cleanup non rollups code

### PR DESCRIPTION
**Pull request by nicklemmon**
[SA-2022](https://sparkpost.atlassian.net/browse/SA-2022)

### What Changed

- Removes surviving references to the `use-metrics-rollup` account-level UI option in both source code and tests

### How To Test

- Test for regressions within the report builder/analytics report feature

### To Do

- [X] Fix unit test failures/remove snapshot tests at are not helpful
- [x] Should we rename/remove the `getRollupsPrecision` helper?

### CODE REVIEW

File: src/actions/reportOptions.js [hunk 1]
```diff
@@ -134,11 +127,10 @@ export function refreshReportOptions(payload) {
   return (dispatch, getState) => {
     const { reportOptions } = getState();
     let update = { ...reportOptions, ...payload };
-    const { useMetricsRollup } = selectFeatureFlaggedMetrics(getState());
-    const getPrecision = useMetricsRollup ? getRollupPrecision : getRawPrecision;
+    const getPrecision = getRollupPrecision;
```
Comment (kchu1337) [inline src/actions/reportOptions.js:130]: Can probably just do import { getRollupPrecision as getPrecision}

File: src/components/datePicker/DatePickerNew.js [hunk 2]
```diff
@@ -102,8 +98,7 @@ const datePickerReducer = (state, { type, payload }) => {
 export function DatePicker(props) {
   const [state, dispatch] = useReducer(datePickerReducer, initialState);
   const { isSelecting, selected, isDatePickerOpen } = state;
-
-  const getPrecision = props.useMetricsRollup ? getRollupPrecision : getRawPrecision;
+  const getPrecision = getRollupPrecision;
```
Comment (kchu1337) [inline src/components/datePicker/DatePickerNew.js:101]: Same here. import { getRollupPrecision as getPrecision}

File: src/components/datePicker/ManualEntryFormNew.js [hunk 3]
```diff
@@ -48,7 +47,7 @@ export function ManualEntryForm(props) {
     }
   }, initialState);
 
-  const getPrecision = props.useMetricsRollup ? getRollupPrecision : getRawPrecision;
+  const getPrecision = getRollupPrecision;
```
Comment (kchu1337) [inline src/components/datePicker/ManualEntryFormNew.js:50]: Same here. import { getRollupPrecision as getPrecision}

File: src/pages/reportBuilder/context/ReportBuilderContext.js [hunk 4]
```diff
@@ -40,11 +35,11 @@ const reducer = (state, action) => {
     }
     case 'UPDATE_REPORT_OPTIONS': {
       const { payload, meta } = action;
-      const { useMetricsRollup, subaccounts } = meta;
+      const { subaccounts } = meta;
       let update = { ...state, ...payload };
-      const getPrecision = useMetricsRollup ? getRollupPrecision : getRawPrecision;
+      const getPrecision = getRollupPrecision;
```
Comment (kchu1337) [inline src/pages/reportBuilder/context/ReportBuilderContext.js:40]: Same here. import { getRollupPrecision as getPrecision}

**Review by kchu1337 — CHANGES_REQUESTED** [review]
Added some feedback. I didn't check that you removed all references, but I left feedback for the ones you did remove.

File: src/actions/reportOptions.js (hunk 1 above)
Comment (nicklemmon) [reply src/actions/reportOptions.js:130]: That's a great idea

File: src/pages/reportBuilder/components/DateTimeSection/tests/DateTimeSection.test.js [hunk 5]
```diff
@@ -16,14 +16,9 @@ describe('DateTimeSection', () => {
     );
   };
 
-  it('should disable fields when not using metrics rollup', () => {
-    const { getByLabelText } = subject({ useMetricsRollup: false });
-    expect(getByLabelText('Precision')).toHaveAttribute('disabled');
-    expect(getByLabelText('Time Zone')).toHaveAttribute('disabled');
-  });
+  it('should not disable fields', () => {
```
[the remark this answers is not in the document]
Comment (nicklemmon) [reply src/pages/reportBuilder/components/DateTimeSection/tests/DateTimeSection.test.js:19]: Yeeah - this isn't really a global component either so we can probably just `rm -rf` and move on.

**Comment by nicklemmon** [comment]
@kchu1337 Ready for another look when you have time

**Review by kchu1337 — APPROVED** [review]

**Pull request merged**